src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
```

Comment:
1) _map_ should **not** be static. Otherwise, the products added from different instances will be shared between them.

2) _map_ should be **private**, so it is not directly acessed outside this class. - Now it can be modified from classes outside our control. Also, it is bad for maintenance: if you wanted to change in the future how to retrieve this map, you will need to change all other classed that are accessing to it. - A **getter** should be provided to access this map from other classes.